homeassistant/components/formula_one/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for Formula 1 integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from . import F1Data
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    if not await F1Data(hass).test_connect():
+        raise CannotConnect
+
+    return {"title": "Formula 1"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Formula 1."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        # this integration should not be setup more than once, hence non-unique id
+        await self.async_set_unique_id("f1")
+        self._abort_if_unique_id_configured()
+
+        if user_input is None:
```

Comment:
If you flip this around:

```
if user_input is not None:
```

You can remove the duplicate async_show_form call